### PR TITLE
[FLINK-24846][streaming] Guava import fix in AsyncWaitOperatorTest

### PR DESCRIPTION
**Pull request by akalash**
## What is the purpose of the change

This PR fixes the import of guava in AsyncWaitOperatorTest

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java [hunk 1]
```diff
@@ -65,12 +65,13 @@
...
 import org.apache.flink.util.TestLogger;
 
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
 import org.hamcrest.Matchers;
 import org.junit.Assert;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.Timeout;
-import org.testcontainers.shaded.com.google.common.collect.Lists;
```
Comment (zentol) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java:73]: How about a checkstyle rule to forbid imports from org.testcontainers.shaded?

Comment (akalash) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java:73]: Yes, I think it makes sense to do so. Do you want to add this rule to this PR?

Comment (pnowojski) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java:73]: Yes, let's do it here that way adding this checkstyle will have a bit more of a context. (checkstyle doesn't have to be backported)

Comment (akalash) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java:73]: I pushed the separate commit with checkstyle changes

**Review by zentol — APPROVED** [review]

**Pull request merged**